### PR TITLE
1872 - Consolidate scattered file/S3 helpers into utils/file_utils.py

### PR DESCRIPTION
**Pull request by MarionJHolloway**
## Description
Trello ticket [#1872](https://trello.com/c/rYnyIhSe/1872-fix-scope-of-file-utils)

Follow-up to ticket 1868, per Gary's review comment on [PR #1462](https://github.com/NMDSdevopsServiceAdm/DataEngineering/pull/1462): "opening a new ticket to combine related file for all file helpers." `split_s3_uri` had ended up duplicated three times because neither existing shared `utils.py` is safe to import from for the other framework's consumers (`utils/utils.py` pulls in `pyspark`/`pydeequ` at module level, `polars_utils/utils.py` pulls in `polars`).

This extracts every duplicated/scattered pure file-or-S3-path-or-CSV helper (`split_s3_uri`, `generate_s3_dir`, `list_s3_parquet_import_dates`, `empty_s3_folder`, `get_s3_objects_list`, `get_file_directory`, `construct_s3_uri`, `construct_destination_path`, `is_csv`, `identify_csv_delimiter`, `read_partial_csv_content`) into one new dependency-light `utils/file_utils.py`, and repoints every consumer to it — the 4 real Glue PySpark ingest jobs, `polars_utils/utils.py`'s own internal use, `polars_utils/validation/actions.py`, `convert_delta_to_full_utils.py`, the CQC delta API download jobs, and `cqc_api`'s `utils.py`/`postcode_matcher.py`. `projects/_01_ingest/utils/utils.py` is deleted entirely (all 8 of its functions moved out, zero other consumers).

Also fixes a dependency gap in every Fargate Dockerfile that `COPY polars_utils` — they now also `COPY utils/file_utils.py`, since `polars_utils/utils.py` imports from it at module level (the same bug class ticket 1868 fixed once already).

## Testing
- [x] Unit tests passing
- [x] Successful Step Function run ([add link](https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/statemachines/view/arn:aws:states:eu-west-2:856699698263:stateMachine:1872-file-utils-Ind-CQC-Filled-Post-Estimates))
- [ ] Outputs checked in Athena

No data pipeline outputs change — this is a pure code-location refactor (same function bodies, same logic), so Step Function/Athena checks aren't applicable. Ran the full non-PySpark-dependent test surface locally (186 tests: new `test_file_utils.py`, `polars_utils` tests, `cqc_api`/`postcode_matcher` tests, `convert_delta_to_full_utils` tests, validation actions tests) — all passing. PySpark-touched test files couldn't run locally (sandbox has no Maven access for the Spark/Deequ fixture) but were verified by `py_compile` and a repo-wide grep for every relocated function's call sites (both `from module import fn` and qualified `alias.fn(...)` patterns); a first CI run caught 3 missed qualified-call sites, which are fixed in the second commit and re-verified by CI.

## Checklist
- [x] Unit tests added/amended
- [x] Docstrings added/updated
- [x] Updated CHANGELOG
- [x] Code reviewed by AI
- [x] Moved Trello ticket to PR column

## Reviewer checklist
- [ ] PR solves the Trello ticket requirements
- [ ] Outputs appear reasonable and understandable
- [ ] The overall approach is appropriate and not over-engineered
- [ ] No obvious scalability, performance or interdependency concerns
- [ ] Tests appropriately cover the main behaviour/edge cases
- [ ] Naming and structure are broadly understandable and consistent
- [ ] Docstrings are sufficient for future maintenance
- [ ] Docstrings cover non-obvious behaviour

## Note for reviewer
AI review flagged one pre-existing, out-of-scope duplication while checking this diff: `read_manual_postcode_corrections_csv_to_dict` is duplicated verbatim between `projects/_01_ingest/cqc_api/utils/utils.py` and `postcode_matcher.py` — not a file/S3 helper, so left untouched here, but worth a follow-up if the team wants it cleaned up.

### CODE REVIEW

File: projects/_01_ingest/ascwds/fargate/Dockerfile [hunk 1]
```diff
@@ -8,6 +8,7 @@ ENV PYTHONPATH=/app
 
 # Dependencies
 COPY polars_utils polars_utils
+COPY utils/file_utils.py utils/file_utils.py
```
Comment (garypolzin) [inline projects/_01_ingest/ascwds/fargate/Dockerfile:11]: Might just be worth running a couple of step functions just to check the Dockerfile situation - when I last tested I made a similar style of filepath to our raw data folder in this branch and put a CSV file in there. The ingest job always fails at some point but so long as it can scan the data in the first place then we at least know the Dockerfile is ok, which is the only thing we need to check here

Comment (MarionJHolloway) [reply projects/_01_ingest/ascwds/fargate/Dockerfile:11]: https://eu-west-2.console.aws.amazon.com/gluestudio/home?region=eu-west-2#/job/1872-file-utils-ingest_ascwds_dataset_job/run/jr_2944fedc0bb21c8a0ff9852f58783a6afcb4d1bd38f56dd04656cf039d659b57

Failed as expected
